File: Tests/SourceSymbolsTests/Fixtures/TypeScript/declarations.ts
// Leading documentation is outside the exported declaration.
export namespace App.Tools {
  export interface Store<T> extends Readonly<T> {
    readonly size: number;
    read(key: string): T;
    read(key: number): T;
  }
  export type Item = { value: string; format(): string };
  export const enum State { Ready, Busy = "busy" }
  @sealed
  export abstract class Client<T extends Item> implements Store<T> {
    static count = 0;
    #secret = "";
    abstract size: number;
    constructor(public value: T) {}
    abstract read(key: string): T;
    read(key: number): T;
    read(key: string | number): T { const local = this.value; return local; }
    get current(): T { return this.value; }
    set current(value: T) { this.value = value; }
    async send(value: T, timeout?: number): Promise<T> { return value; }
  }
  export function choose(value: string): string;
  export function choose(value: number): number;
  export function choose(value: string | number): string | number { return value; }
  export const first = 1, second = 2;
}
declare module "remote" { export function load(): void; }

File: Tests/SourceSymbolsTests/Fixtures/Kotlin/declarations.kt
package shop

// Leading documentation is outside the declaration.
data class Cart<T>(val item: T, private var count: Int = 0) where T : Any {
    constructor(item: T) : this(item, 1) {
        val ready = true
    }
    init {
        val checked = item
    }
    fun add(value: Int) {
        fun local() = value
    }
    fun add(value: String) = value
    val total: Int
        get() {
            val cached = count
            return cached
        }
    companion object {
        fun create() = 1
    }
    object Named {
        val version = 1
    }
}
interface Reader {
    fun read(): String
}
enum class State {
    OPEN,
    CLOSED;
    fun isOpen() = this == OPEN
}
typealias Callback<T> = (T) -> Unit
val top = 1
fun run() {
    val (first, _, second) = triple()
}
